recipe/patches/0004-Windows-shared-libs.patch

```diff
@@ -0,0 +1,588 @@
+         set(CMAKE_INSTALL_RPATH "@loader_path")
+     else()
+@@ -181,6 +186,8 @@ if(CUDA_FOUND)
+         list(APPEND KALDI_CUDA_NVCC_FLAGS "-Xcompiler -fPIC")
+         set(KALDI_CUDA_NVCC_FLAGS "--default-stream=per-thread;-std=c++${CMAKE_CXX_STANDARD}")
+     endif()
++    set(CMAKE_CUDA_STANDARD 17)
++    set(CMAKE_CUDA_STANDARD_REQUIRED TRUE)
+     set(CUDA_NVCC_FLAGS ${KALDI_CUDA_NVCC_FLAGS} ${CUDA_NVCC_FLAGS})
+ 
+     add_definitions(-DHAVE_CUDA=1)
+@@ -285,8 +292,8 @@ add_subdirectory(src/chainbin)
+ add_subdirectory(src/ivectorbin)
+ if(NOT MSVC)
+     add_subdirectory(src/onlinebin)
+-    add_subdirectory(src/online2bin)
+ endif()
++add_subdirectory(src/online2bin)
+ add_subdirectory(src/kwsbin)
+ 
+ # add all cuda executables
+diff --git a/src/base/kaldi-dll-exports.h b/src/base/kaldi-dll-exports.h
+new file mode 100644
+index 0000000..ee1d84c
+--- /dev/null
++++ b/src/base/kaldi-dll-exports.h
+@@ -0,0 +1,29 @@
++
++#ifndef kaldi_dll_export_H
++#define kaldi_dll_export_H
++
++#if defined(KALDI_DLL_EXPORTS)
++#  define KALDI_DLL __declspec(dllexport)
++#elif defined(KALDI_DLL_IMPORTS)
++#  define KALDI_DLL __declspec(dllimport)
++#else
++#  define KALDI_DLL
++#endif // defined(KALDI_DLL_EXPORTS)
++
++#if defined(KALDI_UTIL_DLL_EXPORTS)
++#  define KALDI_UTIL_DLL __declspec(dllexport)
++#elif defined(KALDI_UTIL_DLL_IMPORTS)
++#  define KALDI_UTIL_DLL __declspec(dllimport)
++#else
++#  define KALDI_UTIL_DLL
++#endif // defined(KALDI_UTIL_DLL_EXPORTS)
++
++#if defined(KALDI_CUMATRIX_DLL_EXPORTS)
++#  define KALDI_CUMATRIX_DLL __declspec(dllexport)
++#elif defined(KALDI_CUMATRIX_DLL_IMPORTS)
++#  define KALDI_CUMATRIX_DLL __declspec(dllimport)
++#else
++#  define KALDI_CUMATRIX_DLL
++#endif // defined(KALDI_CUMATRIX_DLL_EXPORTS)
++
++#endif /* kaldi_dll_export_H */
+diff --git a/src/base/kaldi-error.h b/src/base/kaldi-error.h
+index a9904a7..bc950f3 100644
+--- a/src/base/kaldi-error.h
++++ b/src/base/kaldi-error.h
+@@ -33,6 +33,7 @@
+ 
+ #include "base/kaldi-types.h"
+ #include "base/kaldi-utils.h"
++#include "base/kaldi-dll-exports.h"
+ /* Important that this file does not depend on any other kaldi headers. */
+ 
+ #ifdef _MSC_VER
+@@ -54,7 +55,7 @@ void SetProgramName(const char *basename);
+ 
+ /// This is set by util/parse-options.{h,cc} if you set --verbose=? option.
+ /// Do not use directly, prefer {Get,Set}VerboseLevel().
+-extern int32 g_kaldi_verbose_level;
++extern int32 KALDI_DLL g_kaldi_verbose_level;
+ 
+ /// Get verbosity level, usually set via command line '--verbose=' switch.
+ inline int32 GetVerboseLevel() { return g_kaldi_verbose_level; }
+diff --git a/src/base/kaldi-types.h b/src/base/kaldi-types.h
+index 68d5578..3a1b8b9 100644
+--- a/src/base/kaldi-types.h
++++ b/src/base/kaldi-types.h
+@@ -39,16 +39,30 @@ typedef float   BaseFloat;
+ // we find in the future lacks stdint.h
+ #include <stdint.h>
+ 
+-typedef int8_t   int8;
+-typedef int16_t  int16;
+-typedef int32_t  int32;
+-typedef int64_t  int64;
```

Comment:
This part seems to be duplicated with patch 3. Probably it passes because `patch` recognizes it has already been applied...?

Reply:
Yeah I did one patch for the 1.82 changes on the original PR and then another patch on top of those with my changes to get it working, just to keep it separate, but getting rid of the typedefs meant that I undid what was done in the original PR.